bindings/rust/bench/README.md

```diff
@@ -0,0 +1,43 @@
+# Benchmarking s2n-tls
+
+We benchmark s2n-tls against two other common TLS libraries, Rustls and OpenSSL. We use Rust bindings for s2n-tls and OpenSSL and Criterion to benchmark these libraries. 
+
+## Setup 
+
+Setup is easy! Just run `/bindings/rust/generate.sh` to generate Rust bindings for s2n-tls and have OpenSSL installed.
+
+## Running benchmarks
+
+To run our results, just run `cargo bench`! Critierion will auto-generate an HTML report in `target/criterion/`. 
+
+## Future features
+
+For these libraries, in the future we plan to benchmark:
+- bulk throughput with different cipher suites
+- handshakes with client authentication (mTLS)
+- handshakes with different certificate signature algorithms
+- and session resumption.
+
+We will also conduct memory benchmarks for all features and test the historical performance of s2n-tls.
+
+## Implementation details
+
+All of our benchmarks are run in Rust on a single thread for consistency.
+
+### IO
+
+To remove external factors, we use custom IO with our benchmarks, bypassing the networking layer and having the client and server connections transfer data to each other via a local buffer. 
+
+Because s2n-tls Rust bindings only allows the use of a `*mut c_void` context pointer and callbacks to handle IO, we use a `Pin<Box<UnsafeCell<VecDeque<u8>>>>` for our local buffers, getting the pointer to the context `VecDeque` using `UnsafeCell::get()`. We use `Pin<Box<T>>` to ensure the context pointer passed to s2n-tls is constant and remains valid. We use `UnsafeCell` to wrap the context pointer, and the single-threaded nature of the benchmarks ensures only one connection is mutating the context at once.
```

Comment:
Why is this information important for a reader to know?

Reply:
This is just to be transparent with the design decisions made given the limitations of the s2n-tls Rust bindings. I wanted to call out how the custom IO for s2n-tls worked, given that it's nonobvious and very different from Rustls and OpenSSL which use `Read+Write` trait objects.

Reply:
I think it might be worth separating out the nitty-gritty implementation details into a different section. The "we aren't going over the networking stack" is relevant to most readers, but the internal implementation details are less likely to be useful.

Reply:
Would having just the comment in the code be good enough then? I originally wanted to call it out in the readme in case someone would use the decision to use the complex type as a criticism against the benchmarks, but if it's natural enough and there isn't a much better way, we don't need to call it out right?

Reply:
Ya, I'd be fine with the comment in code.